### PR TITLE
S3

### PR DESCRIPTION
**Pull request by CuriousSugam**
- Keep track of the network usage data of every application (both wifi and mobile data)

### CODE REVIEW

File: app/src/main/java/com/lftechnology/activitylogger/ConnectivityChangeBroadcastReceiver.java [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (intent.getAction() != null && intent.getAction().equals("android.net.conn.CONNECTIVITY_CHANGE")) {
+            ConnectivityManager connectivityManager = (ConnectivityManager) context.getSystemService(Context.CONNECTIVITY_SERVICE);
+            NetworkInfo networkInfo = connectivityManager.getActiveNetworkInfo();
+            Intent serviceIntent = new Intent(context, ConnectivityChangeMonitoringIntentService.class);
+            if (networkInfo != null) {
+                if (networkInfo.getType() == ConnectivityManager.TYPE_WIFI && networkInfo.isConnected()) {
+                    serviceIntent.putExtra("networkType", "wifi");
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/ConnectivityChangeBroadcastReceiver.java]: @CuriousSugam define the keys as constants.

**Comment by manas-raj-shrestha** [comment]
@CuriousSugam are you sure that you want to merge this into develop? you mentioned that the splash screen is not functioning properly.

File: app/src/main/AndroidManifest.xml [hunk 2]
```diff
@@ -1,20 +1,42 @@
...
             </intent-filter>
         </activity>
-    </application>
+        <activity android:name=".MainActivity"
+            android:theme="@style/AppTheme">
+        </activity>
+        <receiver android:name=".ConnectivityChangeBroadcastReceiver" >
+            <intent-filter android:priority="999">
+                <action android:name="android.net.conn.CONNECTIVITY_CHANGE" />
+                <!--<category android:name="android.intent.category.DEFAULT"/>-->
```
Comment (manas-raj-shrestha) [inline app/src/main/AndroidManifest.xml]: @CuriousSugam remove the commented lines

File: app/src/main/java/com/lftechnology/activitylogger/CheckPermissions.java [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+     */
+    public static boolean isPermissionForAccessSet(Context context) {
+
+        try {
+            PackageManager packageManager = context.getPackageManager();
+            ApplicationInfo applicationInfo = packageManager.getApplicationInfo(context.getPackageName(), PackageManager.GET_META_DATA);
+            AppOpsManager appOpsManager = (AppOpsManager)context.getSystemService(Context.APP_OPS_SERVICE);
+            int mode = appOpsManager.checkOpNoThrow(AppOpsManager.OPSTR_GET_USAGE_STATS, applicationInfo.uid, applicationInfo.packageName);
+            if(mode != AppOpsManager.MODE_ALLOWED){
+                return false;
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/CheckPermissions.java]: @CuriousSugam if possible use only one return statement in a method

File: app/src/main/java/com/lftechnology/activitylogger/CheckPermissions.java [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+            }else{
+                return true;
+            }
+        } catch (PackageManager.NameNotFoundException e) {
+            return false;
+        }
+
+        //if (RawAppInfo.getUsageStatsAppList(context).isEmpty())
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/CheckPermissions.java]: @CuriousSugam remove comments

File: app/src/main/java/com/lftechnology/activitylogger/ConnectivityChangeBroadcastReceiver.java [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+ * ConnectivityChangeBroadcastReceiver is a broadcast receiver. It receives the broadcast from the
+ * android system when the network connectivity changes.
+ *
+ * Created by Sugam on 7/11/2016.
+ */
+public class ConnectivityChangeBroadcastReceiver extends BroadcastReceiver {
+
+    @Override
+    public void onReceive(Context context, Intent intent) {
+        if (intent.getAction() != null && intent.getAction().equals("android.net.conn.CONNECTIVITY_CHANGE")) {
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/ConnectivityChangeBroadcastReceiver.java]: @CuriousSugam define the intent action name in a constant

File: app/src/main/java/com/lftechnology/activitylogger/RawAppInfo.java [hunk 6]
```diff
@@ -0,0 +1,67 @@
...
+import java.text.SimpleDateFormat;
+import java.util.Calendar;
+import java.util.List;
+/**
+ * Created by sparsha on 6/29/2016.
+ * Returns raw information of apps that are run within an Interval Provided.
+ * Default is set to daily
+ */
+public class RawAppInfo {
+    private static final SimpleDateFormat dateFormat = new SimpleDateFormat();//Gets the Date Format
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/RawAppInfo.java]: @CuriousSugam static final variable are written in upper case DATE_FORMAT

File: app/src/main/java/com/lftechnology/activitylogger/Services/ConnectivityChangeMonitoringIntentService.java [hunk 7]
```diff
@@ -0,0 +1,99 @@
...
+    protected void onHandleIntent(Intent intent) {
+
+        if (intent.hasExtra("networkType")) {
+            String networkType = intent.getStringExtra("networkType");
+            SharedPreferences sharedPreferences = getSharedPreferences("networkState", MODE_PRIVATE);
+
+            if (networkType.equals("wifi")) {    // when wifi is turned on
+                // fill into the temporary table and set the current network type i.e 'wifi' into the shared preference
+                fillIntoNetworkTempTable();
+                sharedPreferences.edit().putBoolean("wifi", true).apply();
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/Services/ConnectivityChangeMonitoringIntentService.java:40]: @CuriousSugam put the key in constants

File: app/src/main/java/com/lftechnology/activitylogger/Controller/SQLiteAccessLayer.java [hunk 8]
```diff
@@ -0,0 +1,318 @@
...
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.ArrayList;
+import java.util.Date;
+import java.util.List;
+
+/**
+ * @author Sugam Shakya
+ * Created by Sugam on 7/5/2016.
+ * This class provides the interface to insert, update, delete and query the app details in the application database
```
Comment (manas-raj-shrestha) [inline app/src/main/java/com/lftechnology/activitylogger/Controller/SQLiteAccessLayer.java]: @CuriousSugam This class provides public methods to insert......

File: app/src/main/AndroidManifest.xml (hunk 2 above)
Comment (CuriousSugam) [inline app/src/main/AndroidManifest.xml]: @manas-raj-shrestha  Receiving broadcasts, starting services and monitoring the network data statistics is a seperate story than splash screen, and a new feature is to be branched and merged to dev, isn't it?

**Pull request merged**